Source/Engine/Engine/RandomUtil.cs

```diff
@@ -24,5 +24,80 @@ public static float Rand()
         {
             return Random.Next(0, int.MaxValue) / (float)int.MaxValue;
         }
+
+        /// <summary>
+        ///Generates a random number within a specified range.
+        /// </summary>
+        /// <param name="min">The minimum value of the range.</param>
+        /// <param name="max">The maximum value of the range.</param>
+        /// <returns>The random number.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float RandRange(float min, float max)
+        {
+            return Rand() * (max - min) + min;
+        }
+
+        /// <summary>
+        ///Generates a random number within a specified range.
+        /// </summary>
+        /// <param name="min">The minimum value of the range.</param>
+        /// <param name="max">The maximum value of the range.</param>
+        /// <returns>The random number.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double RandRange(double min, double max)
+        {
+            return Random.NextDouble() * (max - min) + min;
+        }
+
+        /// <summary>
+        /// Generates a random number within a specified range.
+        /// </summary>
+        /// <param name="min">The minimum value of the range.</param>
+        /// <param name="max">The maximum value of the range.</param>
+        /// <returns>The random number.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int RandRange(int min, int max)
+        {
+            return Random.Next(min, max);
+        }
+
+        /// <summary>
+        /// Generates a random point inside a sphere with radius 1.0.
+        /// </summary>
+        /// <returns>A random point within the sphere.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Float3 InsideSphere()
+        {
+            float u = RandRange(-1, 1); 
+            float v = RandRange(-1, 1); 
+            float w = RandRange(-1, 1); 
+
+            // Normalize the vector to ensure it's inside the unit sphere
+            float length = Mathf.Sqrt(u * u + v * v + w * w);
+            u /= length;
+            v /= length;
+            w /= length;
+
+            return new Float3(u, v, w);
```

Comment:
Doesn't this return a biased sample of random points inside a sphere?

Reply:
Oh, we actually have an unbiased one over there https://github.com/FlaxEngine/FlaxEngine/blob/97ed46bc4ff36a0269b50b138c08908461ee9f96/Source/Engine/Engine/RandomStream.cs#L131